data_steward/retraction/retract_deactivated_pids.py

```diff
@@ -169,430 +143,221 @@ def get_table_dates_info(project_id, dataset_id, client):
+    table_dates_info = get_table_dates_info(table_cols_df)
+    tables = table_cols_df['table_name'].to_list()
+    is_deid = ru.is_deid_dataset(dataset_id)
+    if is_deid and pid_rid_table_ref is None:
+        raise RuntimeError(
+            f"PID-RID mapping table must be specified for deid dataset {dataset_id}"
         )
-        if check_pid_exist(date_row, client, pids_project_id, pids_dataset_id,
-                           pids_table):
-            dataset_list.add(date_row.dataset_id)
-            row = {
-                'project_id': date_row.project_id,
-                'dataset_id': date_row.dataset_id,
-                'table': date_row.table,
-                'date_column': date_row.date_column,
-                'start_date_column': date_row.start_date_column,
-                'end_date_column': date_row.end_date_column
-            }
-            final_date_column_df = final_date_column_df.append(
-                row, ignore_index=True)
-
-    LOGGER.info(
-        "Looping through the deactivated PIDS df to create queries based on the retractions needed per PID table"
-    )
-    for ehr_row in deactivated_ehr_pids_df.itertuples(index=False):
-        LOGGER.info(f'Creating retraction queries for PID: {ehr_row.person_id}')
-        for date_row in final_date_column_df.itertuples(index=False):
-            # Determine if dataset is deid to correctly pull pid or research_id and check if ID exists in dataset or if
-            # already retracted
-            if re.match(DEID_REGEX, date_row.dataset_id):
-                pid = get_research_id(date_row.project_id, date_row.dataset_id,
-                                      ehr_row.person_id, client)
-            else:
-                pid = ehr_row.person_id
-
-            # Get or create sandbox dataset
-            sandbox_dataset = check_and_create_sandbox_dataset(
-                date_row.project_id, date_row.dataset_id)
-
-            # Create queries based on type of date field
-            LOGGER.info(
-                f'Creating Query to retract {pid} from {date_row.dataset_id}.{date_row.table}'
+    sandbox_queries = []
+    clean_queries = []
+    for table in table_dates_info:
+        table_ref = gbq.TableReference.from_string(
+            f"{project_id}.{dataset_id}.{table}")
+        mapping_table = f'_mapping_{table}'
+        ext_table = f'{table}_ext'
+        has_mapping = mapping_table in tables
+        has_ext = ext_table in tables
+        if has_mapping:
+            mapping_ext_ref = gbq.TableReference.from_string(
+                f'{project_id}.{dataset_id}.{mapping_table}')
+        elif has_ext:
+            mapping_ext_ref = gbq.TableReference.from_string(
+                f'{project_id}.{dataset_id}.{ext_table}')
+        elif table == 'death':
+            mapping_ext_ref = None
+        else:
+            raise RuntimeError(
+                f"No mapping or ext tables for {table}, cannot identify EHR data"
```

Comment:
Note to myself:  Is retract deactivated pids supposed to focus on only EHR data?

Reply:
Retracting EHR data is the full scope of this ticket.